### PR TITLE
Add data plane status rpc to mock management plane

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Add data plane status rpc to mock management plane.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: test/mock/grpc/mock_management_plane_grpc_server.go [hunk 1]
```diff
@@ -44,9 +51,23 @@ func NewManagementGrpcServer() *ManagementGrpcServer {
...
+	server.GET("/api/v1/status", func(c *gin.Context) {
+		if mgs.updateDataPlaneStatusRequest == nil {
+			c.JSON(http.StatusNotFound, nil)
+		} else {
+			mgs.updateDataPlaneStatusMutex.Lock()
+			c.JSON(http.StatusOK, gin.H{
+				"updateDataPlaneStatusRequest": mgs.updateDataPlaneStatusRequest,
+			})
+			mgs.updateDataPlaneStatusMutex.Unlock()
 		}
```
Comment (olli-holmala) [inline test/mock/grpc/mock_management_plane_grpc_server.go:71]: If we are gonna lock for responses, should we also lock for the first branch of the if-clause?
```suggestion
        mgs.updateDataPlaneStatusMutex.Lock()
		if mgs.updateDataPlaneStatusRequest == nil {
			c.JSON(http.StatusNotFound, nil)
		} else {
			c.JSON(http.StatusOK, gin.H{
				"updateDataPlaneStatusRequest": mgs.updateDataPlaneStatusRequest,
			})
		}
		mgs.updateDataPlaneStatusMutex.Unlock()
```

File: test/mock/grpc/mock_management_plane_grpc_server.go [hunk 2]
```diff
@@ -44,9 +51,23 @@ func NewManagementGrpcServer() *ManagementGrpcServer {
 		if mgs.connectionRequest == nil {
 			c.JSON(http.StatusNotFound, nil)
 		} else {
+			mgs.connectionMutex.Lock()
 			c.JSON(http.StatusOK, gin.H{
 				"connectionRequest": mgs.connectionRequest,
 			})
+			mgs.connectionMutex.Unlock()
+		}
```
Comment (olli-holmala) [inline test/mock/grpc/mock_management_plane_grpc_server.go:59]: If we are gonna lock for responses, should we also lock for the first branch of the if-clause?
```suggestion
        mgs.connectionMutex.Lock()
		if mgs.connectionRequest == nil {
			c.JSON(http.StatusNotFound, nil)
		} else {
			c.JSON(http.StatusOK, gin.H{
				"connectionRequest": mgs.connectionRequest,
			})
		}
		mgs.connectionMutex.Unlock()
```

Comment (olli-holmala) [reply test/mock/grpc/mock_management_plane_grpc_server.go:59]: Now that I'm better caffeinated, I realize you're locking for the read of `mgs.updateDataPlaneStatusRequest` but the suggestion still stands, as the if-clause does read that field.

Comment (dhurley) [reply test/mock/grpc/mock_management_plane_grpc_server.go:59]: good spot. updating that there now

**Review by olli-holmala — APPROVED** [review]

**Pull request merged**